airgap/terraform/user-data-scripts/script.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# (7) create nginx site configuration to be as proxy for the registry container and reload Nginx configuration
+
+cat << EOF > /etc/nginx/conf.d/registry.conf
+
+upstream docker-registry {
+  server 127.0.0.1:5000;
+}
+
+server {
+    listen 80;
+    server_name ${registry_url};
+
+    include letsencrypt.conf;
+}
+
+server {
+  listen 443;
+  server_name ${registry_url};
+
+  # SSL
+  #ssl_certificate
+  #ssl_certificate_key
+
+  # disable any limits to avoid HTTP 413 for large image uploads
+  client_max_body_size 0;
+
+  # required to avoid HTTP 411: see Issue #1486 (https://github.com/docker/docker/issues/1486)
+  chunked_transfer_encoding on;
+
+  location /v2/ {
+    # Do not allow connections from docker 1.5 and earlier
+    # docker pre-1.6.0 did not properly set the user agent on ping, catch "Go *" user agents
+    if (\$http_user_agent ~ "^(docker\/1\.(3|4|5(?!\.[0-9]-dev))|Go ).*$" ) {
+      return 404;
+    }
+
+    # To add basic authentication to v2 use auth_basic setting plus add_header
+    auth_basic "Require Authentication";
+    auth_basic_user_file /etc/nginx/registry.password;
+    add_header 'Docker-Distribution-Api-Version' 'registry/2.0' always;
+
+    proxy_pass                          http://docker-registry;
+    proxy_set_header  Host              \$http_host;   # required for docker client's sake
+    proxy_set_header  X-Real-IP         \$remote_addr; # pass on real client's IP
+    proxy_set_header  X-Forwarded-For   \$proxy_add_x_forwarded_for;
+    proxy_set_header  X-Forwarded-Proto \$scheme;
+    proxy_read_timeout                  900;
+  }
+}
+EOF
+
+# check and reload Nginx configuration.
+nginx -t
+nginx -s reload
+
+# (8) create letsencrypt certificate.
+
+certbot certonly --webroot -w /var/www/letsencrypt -d ${registry_url} --non-interactive --agree-tos -m yang.chiu@suse.com
+
+# (9) Edit and uncomment the ssl configuration lines in /etc/nginx/conf.d/registry.conf and replace the paths to match the created ssl certificate and key
+
+sed -i "s/listen 443;/listen 443 ssl;/g" /etc/nginx/conf.d/registry.conf
+sed -i "s/#ssl_certificate\b/ssl_certificate \/etc\/letsencrypt\/live\/${registry_url}\/fullchain.pem;/g" /etc/nginx/conf.d/registry.conf
+sed -i "s/#ssl_certificate_key\b/ssl_certificate_key \/etc\/letsencrypt\/live\/${registry_url}\/privkey.pem;/g" /etc/nginx/conf.d/registry.conf
+
+nginx -t
+nginx -s reload
+
+# (10) test registry with docker login command.
+
+docker login -u ${registry_username} -p ${registry_password} ${registry_url}
+
+# (11) pull longhorn images into registry
+
+wget https://raw.githubusercontent.com/longhorn/longhorn/${longhorn_version}/deploy/longhorn-images.txt
+wget https://raw.githubusercontent.com/longhorn/longhorn/v1.2.4/scripts/save-images.sh
```

Comment:
`v1.2.4` this should be configured instead.

Reply:
let it be ${longhorn_version}